### PR TITLE
Optimize docker build and workflows

### LINKED ISSUE #2: Use env variables in GitHub workflows
Not sure if its worth it just noting down since it came to my mind: We have the repo URL in an env variable but we hard code the download URL here. Would it be worth using a variable for the download link as well?

_Originally posted by @dnlggr in https://github.com/joinmarket-webui/joinmarket-webui-docker/pull/1#discussion_r798712935_

### LINKED ISSUE #5: Reuse GitHub workflows
This is pretty much the same as the previous file just with different triggers and env variables, right? 

Not sure if it's worth it but maybe we could DRY this up using a [reusable workflow](https://docs.github.com/en/actions/using-workflows/reusing-workflows)?

_Originally posted by @dnlggr in https://github.com/joinmarket-webui/joinmarket-webui-docker/pull/1#discussion_r798727603_

### PR DESCRIPTION
**Pull request by theborakompanioni**
- Improvement on minimizing size of `standalone` docker image (related to #9)
- Reusable GitHub workflows (Closes #5.)
- Do not use hardcoded values in workflows (Closes #2.)

### CODE REVIEW

File: .github/workflows/create-and-publish-docker-manually.yml [hunk 1]
```diff
@@ -6,65 +6,23 @@ on:
       version:
```
Comment (dnlggr) [inline .github/workflows/create-and-publish-docker-manually.yml:6]: Maybe call this workflow `create-and-publish-docker-dev-manually` or so? Just to make it clear that this is not a manual release but a manual release of the dev image (if I read it correctly, that is 😄 ).

File: .github/workflows/update-dependency-version.yml [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+          LINK: ${{ inputs.repo_url }}
+        uses: peter-evans/create-pull-request@v3
+        with:
+          add-paths: .env
+          branch: ${{ env.NAME }}-updates
+          delete-branch: true
+          commit-message: Update ${{ env.NAME }} to ${{ env.RELEASE_TAG }} (automated change)
+          title: Update ${{ env.NAME }} to ${{ env.RELEASE_TAG }}
+          body: |
+            Updates [${{ env.NAME }}][1] to ${{ env.RELEASE_TAG }}
```
[the remark this answers is not in the document]
Comment (theborakompanioni) [reply .github/workflows/update-dependency-version.yml:69]: Uff.. I'm fixing this for what feels like the hundredth time. Sorry.

File: .github/workflows/update-dependency-version.yml [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+        uses: falti/dotenv-action@v0.2.7
+        with:
+          log-variables: true
+    
+      - name: Prepare repo infos
+        id: repoinfo
+        env:
+          REPO_URL: ${{ inputs.repo_url }}
+        run: |
+          echo ::set-output name=name::$(echo ${REPO_URL#*://*/})
```
[the remark this answers is not in the document]
Comment (theborakompanioni) [reply .github/workflows/update-dependency-version.yml:37]: Will add a comment!

File: .github/workflows/update-dependency-version.yml [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+      - name: Update JoinMarket UI Version
+        if: inputs.current_tag != steps.versions.outputs.release_tag
+        env:
+          ENV_VARNAME: ${{ inputs.env_var_name }}
+          RELEASE_TAG: ${{ steps.versions.outputs.release_tag }}
+        run: |
+          sed -i "s/^$ENV_VARNAME=.*/$ENV_VARNAME=$RELEASE_TAG/g" .env
+
+      - name: Create Pull Request
+        if: inputs.current_tag != steps.versions.outputs.elease_tag
```
[the remark this answers is not in the document]
Comment (theborakompanioni) [reply .github/workflows/update-dependency-version.yml:56]: Stupid me. Thank you for reviewing so thoroughly.

File: .github/workflows/update-dependency-version.yml [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+          echo ::set-output name=name_short::$(echo ${REPO_URL##*://*/})
+
+      - name: Get Latest Releases
+        id: versions
+        env:
+          REPO_NAME: ${{ steps.repoinfo.outputs.name }}
+        run: |
+          echo ::set-output name=release_tag::$(curl -sL "https://api.github.com/repos/${REPO_NAME}/releases/latest" | jq -r ".tag_name")
+
+      - name: Update JoinMarket UI Version
```
[the remark this answers is not in the document]
Comment (theborakompanioni) [reply .github/workflows/update-dependency-version.yml:47]: Renamed to "Update Environment Variable"

File: .github/workflows/create-and-publish-docker-manually.yml (hunk 1 above)
Comment (theborakompanioni) [reply .github/workflows/create-and-publish-docker-manually.yml:6]: Yes, that is what it is doing currently. I want to adapt and possibly remove this when it is not needed any longer. Will rename it!

**Review by dnlggr — APPROVED** [review]
I haven't tested it but the code looks good to me! ✅ ✨ 👏

**Pull request merged**